### PR TITLE
Lellky/fixes for macos linux

### PR DESCRIPTION
**Pull request by lellky**
Hello Piotr!

I would very much want to get in contact with you to help you get support in Home Assistant for your awesome Flexit integration.

This PR is just some updates your can have a look at if you want. I had some troubles getting it to work, and still there is some things that a more experienced python developer need to have a look at.

For example, there is something about a bad file descriptor. Don't know why, maybe a file or socket is not closed somewhere in the BAC0 package or in your code 🤷 

```
an error has occurred: [Errno 9] Bad file descriptor
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.10/site-packages/bacpypes/core.py", line 154, in run
    asyncore.loop(timeout=delta, count=1)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncore.py", line 214, in loop
    poll_fun(timeout, map)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncore.py", line 151, in poll
    r, w, e = select.select(r, w, e, timeout)
OSError: [Errno 9] Bad file descriptor
```

The updated "get local device ip address" code was "inspired" by https://stackoverflow.com/questions/166506/finding-local-ip-addresses-using-pythons-stdlib

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 [build-system]
-requires = ["setuptools>=61.0", "BAC0==22.9.21"]
+requires = [
+    "setuptools>=61.0",
+    "BAC0==22.9.21",
+    "pytz==2022.7.1",
+    "decorator==5.1.1",
```
Comment (piotrbulinski) [inline pyproject.toml:6]: Those new dependencies don't seem to be used anywhere. Are those added on purpose or by accident?

**Comment by piotrbulinski** [comment]
Hi Jonas,

thanks for reaching out. We definitely need to get this fixed. When is the error occurring? Is it when you run the `current_mode.py`?

File: pyproject.toml (hunk 1 above)
Comment (lellky) [reply pyproject.toml:6]: I added them on purpose. When trying to run your code (after doing `python3 -m pip install .`) I get errors saying that there are missing dependencies. But I'm not a python developer, I might have done something wrong.

Furthermore when running your integration in home assistant dev mode I get the same error. Adding the missing requirements requirements.json seems to fix it.

**Comment by lellky** [comment]
It looks very promising though 🥇 

Screenshot 2023-01-16 at 09 56 25

**Pull request merged**